state/txindex/kv/kv.go

```diff
@@ -40,6 +47,138 @@ type TxIndex struct {
 	log log.Logger
 }
 
+func (txi *TxIndex) Prune(retainHeight int64) (int64, int64, error) {
+	// Returns numPruned, newRetainHeight, err
+	// numPruned: the number of heights pruned. E.x. if heights {1, 3, 7} were pruned, numPruned == 3
+	// newRetainHeight: new retain height after pruning
+	// err: error
+
+	lastRetainHeight, err := txi.getIndexerRetainHeight()
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to look up last block indexer retain height: %w", err)
+	}
+	if lastRetainHeight == 0 {
+		lastRetainHeight = 1
+	}
+
+	ctx := context.Background()
```

Comment:
It would be good to add an issue to github to remember to thread this context up in the call chain.

Reply:
I am not sure that I understood what you mean well enough to make it myself

Reply:
Not a big deal... this will anyway have to be done as part of the "big" context threading task.